hyperspy/io_plugins/attolight.py

```diff
@@ -0,0 +1,351 @@
+    with open(path, encoding='windows-1252') as status:
+        for line in status:
+            if 'Horizontal Binning:' in line:
+                binning = int(line[line.find(':') + 1:-1])  # binning = binning status
+            if 'Resolution_X' in line:
+                nx = int(line[line.find(':') + 1:-7])
+                # nx = pixel in x-direction
+            if 'Resolution_Y' in line:
+                ny = int(line[line.find(':') + 1:-7])
+                # ny = pixel in y-direction
+            if 'Real Magnification' in line:
+                FOV = float(line[line.find(':') + 1:-1])
+            if 'Grating - Groove Density:' in line:
+                grating = float(line[line.find(':') + 1:-6])
+            if 'Central wavelength:' in line:
+                central_wavelength_nm = float(line[line.find(':') + 1:-4])
+            if 'Channels:' in line:
+                total_channels = int(line[line.find(':') + 1:-1])
+            if 'Signal Amplification:' in line:
+                amplification = int(line[line.find(':x') + 2:-1])
+            if 'Readout Rate (horizontal pixel shift):' in line:
+                readout_rate = int(line[line.find(':') + 1:-4])
+
+            if 'Exposure Time:' in line:
+                exposure_time_ccd_s = float(line[line.find(':') + 1:-3])
+            if 'HYP Dwelltime:' in line:
+                dwell_time_scan_s = float(line[line.find(':') + 1:-4]) / 1000
+            if 'Beam Energy:' in line:
+                beam_acc_voltage_kv = float(line[line.find(':') + 1:-3]) / 1000
+            if 'Gun Lens:' in line:
+                gun_lens_amps = float(line[line.find(':') + 1:-3])
+            if 'Objective Lens:' in line:
+                obj_lens_amps = float(line[line.find(':') + 1:-3])
+            if 'Aperture:' in line:
+                aperture_um = float(line[line.find(':') + 1:-4])
+            if 'Aperture Chamber Pressure:' in line:
+                chamber_pressure_torr = float(line[line.find(':') + 1:-6])
+            if 'Real Magnification:' in line:
+                real_magnification = float(line[line.find(':') + 1:-3])
+
+    # Correct channels to the actual value, accounting for binning. Get
+    # channels on the detector used (if channels not defined, then assume
+    # its 1024)
+    try:
+        total_channels
+    except:
+        total_channels = attolight_systems[attolight_acquisition_system]['channels']
+    channels = total_channels // binning
+
+    # Return metadata
+    return binning, nx, ny, FOV, grating, central_wavelength_nm, channels, amplification, readout_rate, \
+           exposure_time_ccd_s, dwell_time_scan_s, beam_acc_voltage_kv, gun_lens_amps, obj_lens_amps, aperture_um, \
+           chamber_pressure_torr, real_magnification
+
+
+def _store_metadata(dict_tree, hypcard_folder, md_file_name,
+                    attolight_acquisition_system):
+    """
+    Store metadata in the DictionaryTreeBrowser metadata. Stores
```

Comment:
No need to use the `DictionaryTreeBrowser` here and actually its usage in io plugins will be removed for the split of the io library. Instead, in this case, this is appropriate to use the `mapping` approach to map values of the `original_metadata` attribute of a signal - which contains all possible metadata - to the `metadata` attributes - which contains only standardise keys.

Reply:
I don't fully understand... What `mapping` function? What are the standardised keys?
What would this function return then instead of modifying the DictionartTreeBrowser?

Reply:
There should be no `DictionartTreeBrowser` in the reader and there is no reason for this.

> What `mapping` function? 

`mapping` is a dictionary, see for example https://github.com/hyperspy/hyperspy/blob/30d28d23f074bbccc92ff6a537addb22566d2d1d/hyperspy/io_plugins/digital_micrograph.py#L841

> What are the standardised keys?

https://hyperspy.readthedocs.io/en/latest/user_guide/metadata_structure.html#metadata-structure

Reply:
I have also removed the dependency on DictionaryTreeBrowser. Now the metadata is stored as a dictionary only, not as a DictionaryTree anymore. @ericpre is that correct now?